db/migrate/20200911162327_move_endpoint_authentications_to_application_authentication.rb

```diff
@@ -0,0 +1,46 @@
+class MoveEndpointAuthenticationsToApplicationAuthentication < ActiveRecord::Migration[5.2]
+  def up
+    [cost, subscription_watch].each do |type|
+      # Skip the migration if there is no data to migrate.
+      return true if type.nil?
+
+      ActiveRecord::Base.transaction do
+        Application.where(:application_type_id => type.id).each do |app|
+          source = app.source
+          authentications = source.endpoints.flat_map(&:authentications)
```

Comment:
@lindgrenj6 
We have also `aplication_authentications` table
model `Application`: `has_many :authentications, :through => :application_authentications`

We should review if there is use case for this relation and remove it, if not, it's confusing

Reply:
@rvsia is UI using this M:N relationship somewhere?

Reply:
@slemrmartin Yeah. It connects all applications with their assigned authentications.

Reply:
ok, it means this migration should use the applications_authentications too

Reply:
@slemrmartin I think these relationships should already exist. At least, the UI creates them since the table was introduced.

However, with the latest changes, we have got two 'connections' between an app and auths.

I am kinda discussing this issue here https://issues.redhat.com/browse/RHCLOUD-9526